File: internal/models/announcement.go
package models

import (
	"pixelpunk/pkg/common"

	"gorm.io/gorm"
)

/* Announcement 公告模型 */
type Announcement struct {
	ID        uint            `gorm:"primarykey" json:"id"`
	CreatedAt common.JSONTime `json:"created_at"`
	UpdatedAt common.JSONTime `json:"updated_at"`
	DeletedAt gorm.DeletedAt  `gorm:"index" json:"deleted_at,omitempty"`

	Title     string `gorm:"size:255;not null" json:"title"`            // 标题
	Content   string `gorm:"type:text;not null" json:"content"`         // 内容（Markdown格式）
	Summary   string `gorm:"size:500" json:"summary"`                   // 摘要
	IsPinned  bool   `gorm:"default:false;index" json:"is_pinned"`      // 是否置顶
	Status    string `gorm:"size:20;default:draft;index" json:"status"` // 状态: draft, published, archived
	ViewCount int    `gorm:"default:0" json:"view_count"`               // 浏览次数
	CreatedBy uint   `gorm:"index" json:"created_by"`                   // 创建者ID
}

func (Announcement) TableName() string {
	return "announcements"
}

/* BeforeCreate GORM钩子 - 创建前设置默认值 */
func (a *Announcement) BeforeCreate(tx *gorm.DB) error {
	if a.Status == "" {
		a.Status = "draft"
	}
	return nil
}


func (a *Announcement) IsDraft() bool {
	return a.Status == "draft"
}

func (a *Announcement) IsPublished() bool {
	return a.Status == "published"
}

func (a *Announcement) IsArchived() bool {
	return a.Status == "archived"
}

File: web/src/api/types/announcement.ts
/**
 * 公告模块相关类型定义

/* ==================== 公告基础类型 ==================== */

export interface Announcement {
  id: number
  title: string
  content: string // Markdown格式
  summary: string
  is_pinned: boolean
  status: 'draft' | 'published' | 'archived'
  view_count: number
  created_by: number
  created_at: string
  updated_at: string
}

/* ==================== 请求参数类型 ==================== */

export interface CreateAnnouncementRequest {
  title: string
  content: string // Markdown格式
  summary?: string
  is_pinned?: boolean
  status: 'draft' | 'published' | 'archived'
}

export interface UpdateAnnouncementRequest {
  id: number
  title?: string
  content?: string // Markdown格式
  summary?: string
  is_pinned?: boolean
  status?: 'draft' | 'published' | 'archived'
}

export interface AnnouncementListQuery {
  status?: 'draft' | 'published' | 'archived'
  is_pinned?: boolean
  keyword?: string
  page?: number
  page_size?: number
}

/* ==================== 响应类型 ==================== */

export interface PaginationResponse {
  page: number
  page_size: number
  total: number
  total_page: number
}

export interface AnnouncementListResponse {
  data: Announcement[]
  pagination: PaginationResponse
}

export interface AnnouncementSimple {
  id: number
  title: string
  summary: string
  is_pinned: boolean
  view_count: number
  created_at: string
}

export interface AnnouncementDetail {
  id: number
  title: string
  content: string // Markdown格式
  summary: string
  is_pinned: boolean
  view_count: number
  created_at: string
}

export interface PublicAnnouncementListResponse {
  announcements: AnnouncementSimple[]
  total: number
}

/* ==================== 公告设置类型 ==================== */

export interface AnnouncementSettings {
  announcement_enabled?: boolean
  announcement_drawer_position?: 'left' | 'right'
  announcement_drawer_width?: string | number // 支持 px、vw、% 等单位字符串，或纯数字（默认px）
  announcement_display_limit?: number
  announcement_auto_show_delay?: number // 自动弹窗延迟时间（秒）
}

export type AnnouncementSettingsUpdate = Partial<AnnouncementSettings>
